packages/yew/src/virtual_dom/mod.rs

```diff
@@ -109,6 +109,42 @@ impl fmt::Display for AttrValue {
     }
 }
 
+impl AttrValue {
+    /// Consumes the AttrValue and returns the owned String from the AttrValue whenever possible.
+    /// For AttrValue::Rc the <str> is cloned to String in case there are other Rc or Weak pointers to the
+    /// same allocation.
+    pub fn into_string(self) -> String {
+        match self {
+            AttrValue::Static(s) => (*s).to_owned(),
+            AttrValue::Owned(s) => s,
+            AttrValue::Rc(mut rc) => {
+                if let Some(s) = Rc::get_mut(&mut rc) {
+                    (*s).to_owned()
+                } else {
+                    rc.to_string()
+                }
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests_attr_value {
+    use super::*;
+
+    #[test]
+    fn test_into_string() {
+        let av = AttrValue::Static("str");
+        assert_eq!(av.into_string(), "str");
+
+        let av = AttrValue::Owned("String".to_string());
```

Comment:
```suggestion
        let av = AttrValue::Owned("String".into_string());
```

Reply:
I don't think there is an `into_string()` method for `&static str`:

```no method named `into_string` found for reference `&'static str` in the current scope
method not found in `&'static str`rustcE0599```

Reply:
ah, my bad. Feel free to ignore this.